helm-chart/templates/app.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: workload.crossplane.io/v1alpha1
+kind: KubernetesApplication
+metadata:
+  name: {{ .Release.Name }}-app
+  labels:
+    app: {{ .Release.Name }}
+spec:
+  resourceSelector:
+    matchLabels:
+      app: {{ .Release.Name }}
+  targetSelector:
+    matchLabels:
+      app: {{ .Release.Name }}
```

Comment:
@muvaf I believe this will need to be a conditional because we don't want to match labels if we are just defaulting to an exiting `KubernetesTarget`

Reply:
Thanks for the heads-up! Just fixed it